### PR TITLE
Dremio jdbc plugin

### PR DESCRIPTION
**Pull request by iNikitaGricenko**
### What changes are being made and why?

- Added dreamio query task and Trigger using the JDBC drive.
- Added Query task using the Arrow Flight SQL JDBC drive.
---

### How the changes have been QAed?
Dreamio query was tested through Dremio Test Drive and Dremio OSS (Docker)
 Arrow Flight SQL was tested through Dremio account, adding source that supports Apache Arrow Flight sql 

```yaml
id : dremio-listen
namespace: dev

triggers:
  - id: watch
    type: io.kestra.plugin.jdbc.dremio.Trigger
    url: "jdbc:arrow-flight-sql://localhost:31010/?useEncryption=false"
    username: "dremio"
    password: "dremio123"
    sql: "SELECT * from \"dremio-test\".\"postgres.public\".departments"
    fetch: true
    interval: PT10S

tasks:
  - id: query
    type: io.kestra.plugin.jdbc.dremio.Query
    url: "jdbc:arrow-flight-sql://localhost:31010/?useEncryption=false"
    username: "dremio"
    password: "dremio123"
    sql: "SELECT * from \"dremio-test\".\"postgres.public\".departments"
    fetchOne: true
```
---

### Setup Instructions

- [Dremio Arrow Flight Authentication (URL example)](https://docs.dremio.com/current/sonar/client-applications/drivers/arrow-flight-sql-jdbc-driver/)

### CODE REVIEW

File: plugin-jdbc-dremio/src/main/java/io/kestra/plugin/jdbc/dremio/ArrowFlightQuery.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import java.sql.SQLException;
+import java.time.ZoneId;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Query a Dremio database through Apache Arrow Flight sql driver."
```
Comment (loicmathieu) [inline plugin-jdbc-dremio/src/main/java/io/kestra/plugin/jdbc/dremio/ArrowFlightQuery.java:27]:
```suggestion
    title = "Query a Dremio database through Apache Arrow Flight SQL."
```

File: plugin-jdbc-dremio/src/main/java/io/kestra/plugin/jdbc/dremio/ArrowFlightQuery.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Send a sql query to a Dremio direct database and fetch a row as outputs using Apache Arrow Flight sql driver",
```
Comment (loicmathieu) [inline plugin-jdbc-dremio/src/main/java/io/kestra/plugin/jdbc/dremio/ArrowFlightQuery.java:32]:
```suggestion
            title = "Send a SQL query to a Dremio database and fetch a row as outputs using Apache Arrow Flight SQL",
```

File: plugin-jdbc-dremio/src/main/java/io/kestra/plugin/jdbc/dremio/ArrowFlightQuery.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            code = {
+                "url: jdbc:arrow-flight-sql://localhost:31010/?useEncryption=false",
+                "username: dremio",
+                "password: dremio123",
+                "sql: select * FROM \"postgres.public\".departments",
+                "fetch: true",
+            }
+        ),
+        @Example(
+            title = "Send a sql query to a Dremio coordinator and fetch a row as outputs using Apache Arrow Flight sql driver",
```
Comment (loicmathieu) [inline plugin-jdbc-dremio/src/main/java/io/kestra/plugin/jdbc/dremio/ArrowFlightQuery.java:42]:
```suggestion
            title = "Send a SQL query to a Dremio coordinator and fetch a row as outputs using Apache Arrow Flight SQL",
```

File: plugin-jdbc-dremio/src/main/java/io/kestra/plugin/jdbc/dremio/Query.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Query a Dremio database."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Send a sql query to a Dremio database and fetch a row as outputs",
```
Comment (loicmathieu) [inline plugin-jdbc-dremio/src/main/java/io/kestra/plugin/jdbc/dremio/Query.java:31]:
```suggestion
            title = "Send a SQL query to a Dremio database and fetch a row as outputs",
```

File: plugin-jdbc-dremio/src/main/java/io/kestra/plugin/jdbc/dremio/Trigger.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Wait for query on a Dremio database."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Wait for a sql query to return results and iterate through rows",
```
Comment (loicmathieu) [inline plugin-jdbc-dremio/src/main/java/io/kestra/plugin/jdbc/dremio/Trigger.java:29]:
```suggestion
            title = "Wait for a SQL query to return results and iterate through rows",
```

File: plugin-jdbc-dremio/src/main/java/io/kestra/plugin/jdbc/dremio/Trigger.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+                "    type: io.kestra.core.tasks.flows.EachSequential",
+                "    tasks:",
+                "      - id: return",
+                "        type: io.kestra.core.tasks.debugs.Return",
+                "        format: \"{{json(taskrun.value)}}\"",
+                "    value: \"{{ trigger.rows }}\"",
+                "",
+                "triggers:",
+                "  - id: watch",
+                "    type: io.kestra.plugin.jdbc.clickhouse.Trigger",
```
Comment (loicmathieu) [inline plugin-jdbc-dremio/src/main/java/io/kestra/plugin/jdbc/dremio/Trigger.java:46]:
```suggestion
                "    type: io.kestra.plugin.jdbc.dremio.Trigger",
```

**Review by loicmathieu — CHANGES_REQUESTED** [review]
I don't think using both drivers is useful as you can do the same with the Dremio driver and the Apache Flight SQL driver.

We will discuss internally if we need to change something about that.

**Comment by anna-geller** [comment]
closes https://github.com/kestra-io/kestra/issues/1854

**Pull request closed without merging**